sdk/identity/identity/src/msal/browserFlows/msalAuthCode.ts

```diff
@@ -3,249 +3,352 @@
+  async function doGetToken(
     scopes: string[],
-    options?: CredentialFlowGetTokenOptions,
+    getTokenOptions?: CredentialFlowGetTokenOptions,
   ): Promise<AccessToken> {
-    const account = await this.getActiveAccount();
-    if (!account) {
+    const activeAccount = await getActiveAccount();
+    if (!activeAccount) {
       throw new AuthenticationRequiredError({
         scopes,
-        getTokenOptions: options,
+        getTokenOptions,
         message:
           "Silent authentication failed. We couldn't retrieve an active account from the cache.",
       });
     }
 
     const parameters: msalBrowser.RedirectRequest = {
-      authority: options?.authority || this.msalConfig.auth.authority!,
-      correlationId: options?.correlationId,
-      claims: options?.claims,
-      account: publicToMsal(account),
-      loginHint: this.loginHint,
+      authority: getTokenOptions?.authority || msalConfig.auth.authority!,
+      correlationId: getTokenOptions?.correlationId,
+      claims: getTokenOptions?.claims,
+      account: publicToMsal(activeAccount),
+      loginHint: loginHint,
       scopes,
     };
-    const app = await this.getApp();
-    switch (this.loginStyle) {
+    const msalApp = await getApp();
+    switch (loginStyle) {
       case "redirect":
         // This will go out of the page.
         // Once the InteractiveBrowserCredential is initialized again,
         // we'll load the MSAL account in the constructor.
 
-        await app.acquireTokenRedirect(parameters);
+        await msalApp.acquireTokenRedirect(parameters);
         return { token: "", expiresOnTimestamp: 0, tokenType: "Bearer" };
       case "popup":
-        return this.handleResult(scopes, await app.acquireTokenPopup(parameters));
+        return handleResult(scopes, await app.acquireTokenPopup(parameters));
     }
   }
+
+  /**
+   * Calls to the implementation's doGetToken method.
+   */
+  async function getToken(
+    scopes: string[],
+    getTokenOptions: CredentialFlowGetTokenOptions = {},
+  ): Promise<AccessToken> {
+    const getTokenTenantId =
+      processMultiTenantRequest(tenantId, getTokenOptions, additionallyAllowedTenantIds) ||
+      tenantId;
+
+    if (!getTokenOptions.authority) {
+      getTokenOptions.authority = getAuthority(getTokenTenantId, authorityHost);
+    }
+
+    // We ensure that redirection is handled at this point.
+    await handleRedirect();
+
+    if (!(await getActiveAccount()) && !disableAutomaticAuthentication) {
+      await login(scopes);
+    }
+    return getTokenSilent(scopes, getTokenOptions).catch((err) => {
+      if (err.name !== "AuthenticationRequiredError") {
+        throw err;
+      }
+      if (getTokenOptions?.disableAutomaticAuthentication) {
+        throw new AuthenticationRequiredError({
+          scopes,
+          getTokenOptions,
+          message:
+            "Automatic authentication has been disabled. You may call the authentication() method.",
```

Comment:
nit: I think the method is actually called `authenticate` not `authentication` - I remember seeing the same typo in the node flows so my guess is they probably forgot to rename this when renaming the function or something😄